### PR TITLE
Simplify nav-to implementation.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Followup to https://github.com/dotnet/roslyn/pull/57514

We were tracking what type of search was done so we could decide what to do if no results were found.  But we dont' need to track this information through as we already know what kind of search was done since we are the ones asking.

### CODE REVIEW

File: src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.cs [hunk 1]
```diff
@@ -41,17 +41,15 @@ internal abstract partial class AbstractNavigateToSearchService : INavigateToSea
             };
         }
 
-        public async Task<NavigateToSearchLocation> SearchDocumentAsync(Document document, string searchPattern, IImmutableSet<string> kinds, Func<INavigateToSearchResult, Task> onResultFound, bool isFullyLoaded, CancellationToken cancellationToken)
+        public async Task SearchDocumentAsync(Document document, string searchPattern, IImmutableSet<string> kinds, bool isFullyLoaded, Func<INavigateToSearchResult, Task> onResultFound, CancellationToken cancellationToken)
```
Comment (ryzngard) [inline src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.cs:44]:
```suggestion
        public async Task SearchDocumentAsync(
        	Document document,
      		string searchPattern,
      		IImmutableSet<string> kinds,
      		bool isFullyLoaded,
      		 Func<INavigateToSearchResult, Task> onResultFound,
      		 CancellationToken cancellationToken)
```

File: src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.cs [hunk 2]
```diff
@@ -82,17 +80,15 @@ public async Task<NavigateToSearchLocation> SearchDocumentAsync(Document documen
                 document, searchPattern, kinds, onItemFound, cancellationToken).ConfigureAwait(false);
         }
 
-        public async Task<NavigateToSearchLocation> SearchProjectAsync(Project project, ImmutableArray<Document> priorityDocuments, string searchPattern, IImmutableSet<string> kinds, Func<INavigateToSearchResult, Task> onResultFound, bool isFullyLoaded, CancellationToken cancellationToken)
+        public async Task SearchProjectAsync(Project project, ImmutableArray<Document> priorityDocuments, string searchPattern, IImmutableSet<string> kinds, bool isFullyLoaded, Func<INavigateToSearchResult, Task> onResultFound, CancellationToken cancellationToken)
```
Comment (ryzngard) [inline src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.cs:83]:
```suggestion
        public async Task SearchProjectAsync(
        	Project project,
        	ImmutableArray<Document> priorityDocuments,
        	string searchPattern,
        	IImmutableSet<string> kinds,
        	bool isFullyLoaded,
        	Func<INavigateToSearchResult, Task> onResultFound,
        	CancellationToken cancellationToken)
```

Comment (ryzngard) [reply src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.cs:83]: and other places...

**Review by ryzngard — APPROVED** [review]

File: src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.cs:83]: going to rollt his into the followup PR.

File: src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.cs:44]: going to rollt his into the followup PR.

**Pull request merged**